src/cli/tc.rs

```diff
@@ -193,3 +217,30 @@ pub fn display_schema(blueprint: &Blueprint) {
     let sdl = blueprint.to_schema();
     Fmt::display(format!("{}\n", print_schema::print_schema(sdl)));
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn usage_enabled_true() {
+        assert!(usage_tracking_inner(true, Some(true)));
+        assert!(usage_tracking_inner(false, Some(true)));
+    }
+
+    #[test]
+    fn usage_enabled_false() {
+        assert!(!usage_tracking_inner(true, Some(false)));
+        assert!(!usage_tracking_inner(false, Some(false)));
+    }
+
+    #[test]
+    fn usage_enabled_none_is_prod_true() {
+        assert!(usage_tracking_inner(true, None));
+    }
+
+    #[test]
+    fn usage_enabled_none_is_prod_false() {
+        assert!(!usage_tracking_inner(false, None));
+    }
+}
```

Comment:
We are making `tc` complicated. The concern to track or can be moved to `tailcall-tracker`. 

In `tc` we should create a static instance of the tracker and dispatch events without thinking much. 

`tailcall-tracker` should perform a check to see, if an event should be sent or not.